dace/codegen/prettycode.py

```diff
@@ -44,9 +44,21 @@ def write(self, contents, sdfg=None, state_id=None, node_id=None):
         for line in lines:
             opening_braces = line.count('{')
             closing_braces = line.count('}')
+
+            # Count closing braces before opening ones (e.g., for "} else {")
+            initial_closing_braces = 0
+            for char in line:
```

Comment:
I know this probably isn't performance critical, but... Looping over every character?

Reply:
ok, we can find the first `{` and count the `}` that come in the substring before it...

Reply:
Done.